src/riscv/lib/src/machine_state/memory/state.rs

```diff
@@ -17,6 +21,14 @@ use crate::state_backend::ManagerRead;
 use crate::state_backend::ManagerReadWrite;
 use crate::state_backend::ManagerWrite;
 
+fn get_page_range(address: Address, length: usize) -> Range<usize> {
+    let address = address as usize;
+    let start_page = address >> super::OFFSET_BITS;
+    let end_page =
+        (address.wrapping_add(length).saturating_sub(1) >> super::OFFSET_BITS).saturating_add(1);
+    start_page..end_page
+}
```

Comment:
Nit: I would expect this function along side the generic memory functions in the `memory` module. Would you mind moving it there?